src/sentry/tasks/post_process.py

```diff
@@ -1401,6 +1401,39 @@ def check_has_high_priority_alerts(job: PostProcessJob) -> None:
         )
 
 
+def link_event_to_user_report(job: PostProcessJob) -> None:
+    from sentry.models.userreport import UserReport
+
+    event = job["event"]
+    project = event.project
+    group = event.group
+
+    if features.has(
+        "organizations:user-feedback-event-link-ingestion-changes", project.organization
+    ):
+        metrics.incr("event_manager.save._update_user_reports_with_event_link")
+        event = job["event"]
+        project = event.project
+        user_reports_without_group = UserReport.objects.filter(
+            project_id=project.id,
+            event_id=event.event_id,
+            group_id__isnull=True,
+            environment_id__isnull=True,
+        )
+
+        user_reports_updated = user_reports_without_group.update(
+            group_id=group.id, environment_id=event.get_environment().id
+        )
+
+        if user_reports_updated:
+            metrics.incr("event_manager.save._update_user_reports_with_event_link_updated")
+
+    else:
+        UserReport.objects.filter(project_id=project.id, event_id=job["event"].event_id).update(
+            group_id=group.id, environment_id=event.get_environment().id
+        )
```

Comment:
If overwriting all is fine, then why not just do that instead of trying to get null ones? 
AFAIK an update to same value as already in row should be a no-op in PostgreSQL.

Reply:
that is what we originally did, but am about to make an update where we don't simply want to overwrite all, as we want only the reports that do not currently have a group associated https://github.com/getsentry/sentry/pull/70200